util/settings/settings.go

```diff
@@ -1436,6 +1437,9 @@ func (mgr *SettingsManager) SaveSettings(settings *ArgoCDSettings) error {
 	}
 
 	return mgr.updateSecret(func(argoCDSecret *apiv1.Secret) error {
+		if len(argoCDSecret.Data) > 0 {
```

Comment:
Currently, according to the FAQ, to reset the password you can clear out the admin.password field.  This would require all password resets to generate a new hash out-of-band.

Reply:
You are right, it may not be easy to fix it elegantly. I will explain the reasons for this bug below.

Reply:
> Currently, according to the FAQ, to reset the password you can clear out the admin.password field.  This would require all password resets to generate a new hash out-of-band.

Maybe we can add some annotations to the secret to record the update time and stop updating secret again if it has been updated within the last 30 seconds.